platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -236,6 +258,103 @@ private Response getSamlpRedirectLogoutRequest(String relayState, LogoutRequest
     return ok.build();
   }
 
+  private String extractSubject(HttpSession httpSession) {
+    return Stream.of(httpSession.getAttribute(SecurityConstants.SAML_ASSERTION))
+        .filter(SecurityTokenHolder.class::isInstance)
+        .map(SecurityTokenHolder.class::cast)
+        .map(SecurityTokenHolder::getRealmTokenMap)
+        .map(Map::values)
+        .flatMap(Collection::stream)
+        .map(this::extractSubject)
+        .filter(Objects::nonNull)
+        .map(SubjectUtils::getName)
+        .findFirst()
+        .orElse(null);
+  }
+
+  private Subject extractSubject(SecurityToken securityToken) {
+    try {
+      return securityManager.getSubject(securityToken);
+    } catch (SecurityServiceException e) {
+      LOGGER.debug("Error extracting subject from security token", e);
+      return null;
+    }
+  }
+
+  @POST
+  @Consumes({"text/xml", "application/soap+xml"})
+  public Response soapLogoutRequest(InputStream body, @Context HttpServletRequest request) {
+    XMLObject xmlObject;
+    try {
+      String bodyString = IOUtils.toString(body, StandardCharsets.UTF_8);
```

Comment:
Do we normally do any checking of the input to make sure someone isn't sending us gigabytes worth of data?

Reply:
We currently do not, but I opened a ticket to do just that. Hoping to prioritize the work for next sprint - https://codice.atlassian.net/browse/DDF-3545